src/lib/stores.ts

```diff
@@ -63,24 +96,23 @@ export const handleSnsChange = async (name?: string, principal?: string) => {
 	if (!fetchedCanisters) return;
 
 	sns.reset();
-
+	inputAmount.reset();
 	if (name && principal) {
 		sns.setName(name);
 		const p = Principal.fromText(principal);
 		sns.setPrincipal(principal);
-		const account = await fetchedCanisters.boomerang.get_staking_account(
-			Principal.fromText(principal)
-		);
-		const hex = encodeIcrcAccount({
+		const [account, icpBalanceE8s, nicpBalanceE8s] = await Promise.all([
+			fetchedCanisters.boomerang.get_staking_account(p),
+			fetchIcpBalance(p, fetchedCanisters.icpLedger),
+			fetchNicpBalance(p, fetchedCanisters.nicpLedger)
+		]);
```

Comment:
Here you're tied down to the slowest one, meaning that you'll display balances only when you have the staking account. You should Promise together the balance fetching, but use https://www.geeksforgeeks.org/javascript-fetch-method/ to fetch the staking account on its side.

Reply:
I think from the user perspective, it's fine to have the three of them displayed at the same time even if its tied down to the slowest one. Because otherwise, you might want to make calls or try to use bommerang without every info.